.github/workflows/main.yml

```diff
@@ -23,6 +23,9 @@ jobs:
       - uses: actions/checkout@v1
         with:
           submodules: true
+      - name: Set up Visual Studio shell
+        uses: egor-tensin/vs-shell@v1
```

Comment:
I'm not familiar with this space, but it's surprising to have a dependency on a third-party action that seems to be only 5 days old, to do something which seems like it should be fairly common. Is this the best way to do this?

If so, could you [add a SHA hash](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions)?

Reply:
Last time I looked at how to do this it was a lot of manual work to call out to the vcvars script, which is why I wasn't able to spend the time to figure it out.  I was very pleasantly surprised to discover this action that did the right setup, and did it in a way that made the tools visible in all subsequent shells. :)

Reply:
(I've also audited the action code -- it does Good Stuff (tm).  Added the sha hash of the v1 commit)